lib/config/generator.js

```diff
@@ -0,0 +1,210 @@
+        var presetNames = Object.keys(checker.getConfiguration().getRegisteredPresets());
+        var errorsCollection;
+
+        console.log('Checking', _path, 'against the presets');
+
+        return Vow
+        .all(presetNames.map(this._checkAgainstPreset.bind(this, _path)))
+        .then(function generateErrorsCollection(results) {
+            errorsCollection = [].concat.apply([], results);
+            return errorsCollection;
+        })
+        .then(this._showErrorCounts.bind(this, presetNames))
+        .then(this._getUserPresetChoice.bind(this, prompts[0]))
+        .then(function showViolatedRules(choiceObj) {
+            var presetIndex = choiceObj[prompts[0].name] - 1;
+            var presetName = presetNames[presetIndex];
+
+            console.log('You chose the ' + presetName + ' preset');
+
+            this._config.preset = presetName;
+
+            var errorList = errorsCollection[presetIndex].getErrorList();
+            errorList = getUniqueErrorNames(errorList);
+
+            var violatedRuleCount = errorList.length;
+
+            if (!violatedRuleCount) { return this._config; }
+
+            console.log(_path + ' violates ' + violatedRuleCount + ' rule' + (violatedRuleCount > 1 ? 's' : ''));
+
+            var errorPrompts = generateRuleHandlingPrompts(errorList);
+
+            return this._getUserViolationChoices(errorPrompts)
+            .then(this._handleViolatedRules.bind(this, errorPrompts, errorList))
+            .then(function() {
+                return this._config;
+            }.bind(this));
+        }.bind(this))
+        .then(function flushConfig(config) {
+            fs.writeFileSync(process.cwd() + '/.jscsrc', JSON.stringify(this._config, null, '\t'));
+            console.log('Generated a .jscsrc configuration file in ' + process.cwd());
+        }.bind(this));
+    },
+
+    /**
+     * @private
+     * @param  {String[]} presetNames
+     * @param  {Object[]} errorsCollection
+     */
+    _showErrorCounts: function(presetNames, errorsCollection) {
+        var errorCounts = errorsCollection.map(function(error) {
+            return error.getErrorCount();
+        });
+
+        var table = getTable();
+
+        presetNames.forEach(function(name, idx) {
+            table.push([idx + 1, name, errorCounts[idx]]);
+        });
+
+        console.log(table.toString());
+    },
+
+    /**
+     * Prompts the user to choose a preset
+     *
+     * @private
+     * @param {Object} prompt
+     * @return {Promise}
+     */
+    _getUserPresetChoice: function(prompt) {
+        return this._showPrompt(prompt);
+    },
+
+    /**
+     * Prompts the user to nullify rules or fix violations themselves
+     *
+     * @private
+     * @param  {Object[]} errorPrompts
+     * @return {Promise}
+     */
+    _getUserViolationChoices: function(errorPrompts) {
+        return this._showPrompt(errorPrompts);
+    },
+
+    /** @private */
+    _showPrompt: vowNode.promisify(prompt.get.bind(prompt)),
```

Comment:
A minor annoyance that I have is rolling my own functions and writing tests for them only to have it suggested that I scrap that effort and use a 3rd party solution – or the reverse (using 3rd party and asked to roll my own). 

Is a way to avoid the deletion of work to keep functions in closed scope and during a review, have other maintainers suggest the extraction and testing of the function if they know of other places where it could be applied?